x-pack/plugins/security_solution/server/lib/detection_engine/signals/mappings/build_risk_score_from_mapping.test.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { sampleDocNoSortId } from '../__mocks__/es_results';
+import { buildRiskScoreFromMapping } from './build_risk_score_from_mapping';
+
+describe('buildRiskScoreFromMapping', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('risk score defaults to provided if mapping is incomplete', () => {
+    const riskScore = buildRiskScoreFromMapping({
+      doc: sampleDocNoSortId(),
+      riskScore: 57,
+      riskScoreMapping: undefined,
+    });
+
+    expect(riskScore).toEqual({ riskScore: 57, riskScoreMeta: {} });
+  });
+
+  // TODO: Enhance...
```

Comment:
![](https://media3.giphy.com/media/3ohc14lCEdXHSpnnSU/giphy.gif)

Reply:
Yeah, there are _plenty_ of tests to write for exercising all the areas around building these mappings. I think I saw a situation where the io-ts type wasn't validating correctly and was getting risk scores > 100. So more to dig into here for sure. 🙂